airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresUtils.java

```diff
@@ -30,19 +31,40 @@ public static boolean isCdc(final JsonNode config) {
     return isCdc;
   }
 
+  public static Optional<Integer> getFirstRecordWaitSeconds(final JsonNode config) {
+    final JsonNode replicationMethod = config.get("replication_method");
+    if (replicationMethod != null && replicationMethod.has("initial_waiting_seconds")) {
+      final int seconds = config.get("replication_method").get("initial_waiting_seconds").asInt();
+      return Optional.of(seconds);
+    }
+    return Optional.empty();
+  }
+
+  public static void checkFirstRecordWaitTime(final JsonNode config) {
+    final Optional<Integer> firstRecordWaitSeconds = getFirstRecordWaitSeconds(config);
+    if (firstRecordWaitSeconds.isPresent() && firstRecordWaitSeconds.get() > MAX_FIRST_RECORD_WAIT_TIME.getSeconds()) {
+      throw new IllegalArgumentException(String.format(
+          "Initial waiting seconds cannot be larger than %d seconds for safety.",
+          MAX_FIRST_RECORD_WAIT_TIME.getSeconds()));
+    }
+  }
+
   public static Duration getFirstRecordWaitTime(final JsonNode config) {
-    if (config.has("initial_waiting_seconds")) {
-      final int seconds = config.get("initial_waiting_seconds").asInt();
-      if (seconds > MAX_FIRST_RECORD_WAIT_TIME.getSeconds()) {
+    Duration firstRecordWaitTime = DEFAULT_FIRST_RECORD_WAIT_TIME;
+
+    final Optional<Integer> firstRecordWaitSeconds = getFirstRecordWaitSeconds(config);
+    if (firstRecordWaitSeconds.isPresent()) {
```

Comment:
If `firstRecordWaitSeconds` is 0, what is the expected behavior? From what I can tell there's nothing stopping the user from having `initial_waiting_seconds` to be "0" string, right?

Reply:
Good point.

We set this to 5 seconds in our tests. So previously we did not set a min value for this field. Now this field is included in the check command, I will add the min value check there to prevent users from setting a waiting time that is too short.

Reply:
Done.